### ISSUE TITLE
[Animation System] Rotation around specified origin does not work

### ISSUE THREAD

**Issue by blay09** [MinecraftForge/MinecraftForge#2734]
I've tried to use the Model Animation System to animate the opening/closing of a door for one of my blocks. There doesn't seem to be a way to set the origin within the animation clips (like you can with the axis), so I added a rotation tag with an angle of 0 to the part I'm rotating instead, which does seem to be loaded [here](https://github.com/MinecraftForge/MinecraftForge/blob/master/src/main/java/net/minecraftforge/client/model/animation/ModelBlockAnimation.java#L387).

```
{
    "from": [ 2, 0, 1 ],
    "to": [ 14, 11, 2 ],
    "rotation": { "origin": [0, -8, 8], "axis": "y", "angle": 0 },
    "faces": { ... }
}
```

However, the result of that is this:

![origin](https://cloud.githubusercontent.com/assets/1933180/14362873/d564ebda-fd01-11e5-96dc-170311840425.gif)

It seems to be translating by the origin, rotating, but then not translating back like it's normally done when rotating around a non-centered point. Note how the door is rendering correctly without an offset in the unanimated inventory.

I'm pretty much using the AnimationModelDebug code with some values changed at the moment, since I was just getting started with the new system. Oh, and this is on 1.9.

[Animation State Machine](http://pastebin.com/SPEWrE1r)
[Armature](http://pastebin.com/9ynUMuXx)
+ default AnimationTESR & AnimationStateMachine.load()

**Comment by mincrmatt12** [comment]
Any updates on this? I'm trying to make some complex blocks with this for a new mod, but there still doesn't seem to be a way to set animation origins. Judging by how it's implemented, we either need an extension to TRSRTransformation to allow setting an origin point for the rotations, or a second implementation of ITransformation for setting origin points. The former solution would be simpler, as it only requires modifying TRSRTransformation and adding some extra animation variables

**Comment by mezz** [comment]
`TRSRTransformation` is a final class, so unless I'm misunderstanding the issue you should be able to create a PR to add a method to `TRSRTransformation` without breaking anything.

If you are interested in investigating and fixing this issue, I am happy to help you navigate the PR process.

**Comment by Madsthunder** [comment]
This can be closed (#3875)
